wandb/sdk/interface/artifacts.py

```diff
@@ -165,23 +165,50 @@ def ref_url(self) -> str:
         raise NotImplementedError
 
 
-class ArtifactNotLoggedError(Exception):
-    """Raised for Artifact methods or attributes that require the logged artifact."""
+class ArtifactStatusError(AttributeError):
+    """Raised when an artifact is in an invalid state for the requested operation."""
```

Comment:
(question) what are some invalid states that exist? Would these be artifacts that are finalized? I'm trying to understand if there should be a suggestion in there on moving the artifact to a valid state for the operation to work, or be more specific as to what kinds of invalid states might be causing the error.